src/components/ShoppingList/ShoppingList.js

```diff
@@ -9,42 +9,18 @@ import { visuallyHidden } from '../../style/common.module.css';
 import styles from './styles.module.css';
 
 const ShoppingList = () => {
-  const [items, setItems] = useItems();
+  const [items, actions] = useItems();
   const [text, setText] = useState('');
 
   const addItem = (e) => {
     e.preventDefault();
 
     if (text !== '') {
-      const newItem = {
-        text,
-        key: Date.now(),
```

Comment:
Not worth worrying about in this PR, but are we changing the way this key is set? I seem to remember you mentioning a drawback with this method if a user makes new items quickly.

Reply:
So we now generate the key differently depending on where the item comes from:

- If the item comes from the API then we already have an ID so we use that as the key.
- If the item is added locally then `Date.now()` is used which has millisecond precision - since the user won't be able to manually add more than one item in a millisecond this is pretty safe.